src/app/components/splitter/splitter.ts

```diff
@@ -158,16 +158,13 @@ export class Splitter {
     }
 
     ngAfterContentInit() {
-        this.templates.forEach((item) => {
-            switch (item.getType()) {
-                case 'panel':
-                    this.panels.push(item.template);
-                    break;
-                default:
-                    this.panels.push(item.template);
-                    break;
-            }
+
+        this.templates.changes.subscribe(_ => {
+            this.initPanels();
+            this.cd.detectChanges();
         });
+
+        this.initPanels();
```

Comment:
what about this variant and remove `this.initPanels();` outside pipe.

```ts
this.templates.changes.pipe(
  startWith(null),
  ...unsubscribe
).subscribe(() => {
 this.initPanels();
 this.cd.markForCheck();
})
```

Reply:
yes, it's better but unsubscribe should go into ngOnDestroy to preserve the dynamicity. I'll check it for a new commit.